File: Source/Aladdin.CAPI.STB.Java/src/aladdin/capi/stb/stb11762/BDSBDHPublicKey.java
package aladdin.capi.stb.stb11762;
import java.math.*; 

///////////////////////////////////////////////////////////////////////
// Открытый ключ подписи и обмена
///////////////////////////////////////////////////////////////////////
public class BDSBDHPublicKey extends aladdin.capi.PublicKey implements IBDSBDHPublicKey
{
    // номер версии для сериализации
    private static final long serialVersionUID = -4102993374421267783L;

    // параметры ключа
    private final IBDSBDHParameters parameters; 
    // открытый ключ подписи и обмена
	private final BigInteger bdsY; private final BigInteger bdhY;	
    
    // конструктор
	public BDSBDHPublicKey(aladdin.capi.KeyFactory keyFactory, 
        IBDSBDHParameters parameters, BigInteger bdsY, BigInteger bdhY) 
    {
        // сохранить переданные параметры
        super(keyFactory); this.parameters = parameters; this.bdsY = bdsY; this.bdhY = bdhY;
    }
    // параметры ключа
	@Override public final IBDSBDHParameters parameters () { return parameters; }
    // параметры ключа
	@Override public final IBDSBDHParameters getParams() { return parameters; }
    // значение открытого ключа
    @Override public final BigInteger bdsY() { return bdsY; }
    @Override public final BigInteger getY() { return bdsY; }
    // значение открытого ключа
    @Override public final BigInteger bdhY() { return bdhY; }
}

File: Source/Aladdin.CAPI.STB/CAPI/STB/STB11762/BDSBDHPublicKey.cs
///////////////////////////////////////////////////////////////////////
// Открытый ключ подписи и обмена
///////////////////////////////////////////////////////////////////////
namespace Aladdin.CAPI.STB.STB11762
{
    public class BDSBDHPublicKey : CAPI.PublicKey, IBDSBDHPublicKey
    {
        // параметры ключа
        private IBDSBDHParameters parameters; 
        // открытый ключ подписи и обмена
	    private Math.BigInteger bdsY; private Math.BigInteger bdhY;	
    
        // конструктор
	    public BDSBDHPublicKey(CAPI.KeyFactory keyFactory, IBDSBDHParameters parameters, 
            Math.BigInteger bdsY, Math.BigInteger bdhY) : base(keyFactory)
        {
            // сохранить переданные параметры
            this.parameters = parameters; this.bdsY = bdsY; this.bdhY = bdhY;
        }
        // параметры ключа
	    public override CAPI.IParameters Parameters { get { return parameters; }}
        // значение открытого ключа
	    Math.BigInteger IBDSPublicKey.Y { get { return bdsY; }}
	    Math.BigInteger IBDHPublicKey.Y { get { return bdhY; }}
    }
}
